src/smoothing/SmoothingAndStep.java

```diff
@@ -80,6 +80,10 @@ public double smoothing(int[] params, String inPath, String outPath,
 
                 if((strLine = br.readLine()) != null){
                     while(!addLine(strLine)){
+                        if(Thread.currentThread().isInterrupted()){
```

Comment:
Doing this check in the processing loop feels suboptimal. What was the problem with the external process solution again?